### PR TITLE
fix: SAML login not working on mobile app

### LINKED ISSUE #30049: [6.3.0] SAML login breaks in mobile app
### Description:

When logging in with the mobile app via a SAML IdP, the app logs in but stays in the SSO browser widget. The app itself does not register a successful login. Instead, the Rocket.Chat website is shown in the browser widget.

### Steps to reproduce:

1. Configure SAML IdP
2. Log in via native app
3. App stays in SSO widget, even after successful login

### Expected behavior:

The app should register the successful login and return to its native chat options

### Actual behavior:

image

### Server Setup Information:

- Version of Rocket.Chat Server: 6.3.0
- Operating System: Debian 11
- Deployment Method: tar
- Number of Running Instances: 1
- DB Replicaset Oplog: 1
- NodeJS Version: v14.21.3
- MongoDB Version: 5.0.19

### Client Setup Information

- Desktop App or Browser Version: Android 4.39.0.43604
- Operating System: Android 13

### Additional context

N/A

### Relevant logs:

The app sends a request to `/api/v1/login` with a JSON body:
```json
{
"saml": true,
"credentialToken": "MySecret"
}
```
The server answers with HTTP 401 Unauthorized and the following message:
```json
{
"status": "error",
"message": "No matching login attempt found",
"error": 12345678
}
```
The message obviously comes from https://github.com/RocketChat/Rocket.Chat/blob/856c23555a43853165ba9d9daeed508ccf4c6892/apps/meteor/app/meteor-accounts-saml/server/loginHandler.ts#L23

### PR DESCRIPTION
**Pull request by pierre-lehnen-rc**
## Proposed changes (including videos or screenshots)

Backwards compatibility with mobile app.

## Issue(s)

Closes #30049

### CODE REVIEW

**Review by tapiarafael — APPROVED** [review]

**Review by matheusbsilva137 — DISMISSED** [review]

**Review by sampaiodiego — CHANGES_REQUESTED** [review]
any way of adding some tests to this to prevent further regressions?

File: apps/meteor/app/meteor-accounts-saml/server/lib/SAML.ts [hunk 1]
```diff
@@ -430,7 +430,7 @@ export class SAML {
 				};
 
 				await this.storeCredential(credentialToken, loginResult);
-				const url = `${Meteor.absoluteUrl('saml')}/${credentialToken}`;
+				const url = `${Meteor.absoluteUrl('saml')}/${credentialToken}?saml_idp_credentialToken=${credentialToken}`;
```
Comment (sampaiodiego) [inline apps/meteor/app/meteor-accounts-saml/server/lib/SAML.ts:433]: it is never too late:
```suggestion
				const url = Meteor.absoluteUrl(`saml/${credentialToken}?saml_idp_credentialToken=${credentialToken}`);
```

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

**Review by tapiarafael — APPROVED** [review]

**Comment by pierre-lehnen-rc** [comment]
> any way of adding some tests to this to prevent further regressions?

We're working on getting the CI SAML environment ready for such tests. No test on the core could have caught this specific regression though.

File: apps/meteor/tests/unit/app/meteor-accounts-saml/server.tests.ts [hunk 2]
```diff
@@ -964,4 +964,13 @@ describe('SAML', () => {
 			});
 		});
 	});
+
+	describe('[Utils]', () => {
+		it('should return correct validation action redirect path', () => {
+			const credentialToken = SAMLUtils.generateUniqueID();
+			expect(SAMLUtils.getValidationActionRedirectPath(credentialToken)).to.be.equal(
+				`saml/${credentialToken}?saml_idp_credentialToken=${credentialToken}`,
```
Comment (pierre-lehnen-rc) [inline apps/meteor/tests/unit/app/meteor-accounts-saml/server.tests.ts:972]:
```suggestion
				// the saml_idp_credentialToken param is needed by the mobile app
				`saml/${credentialToken}?saml_idp_credentialToken=${credentialToken}`,
```

File: apps/meteor/app/meteor-accounts-saml/server/lib/Utils.ts [hunk 3]
```diff
@@ -130,6 +130,10 @@ export class SAMLUtils {
 		return newTemplate;
 	}
 
+	public static getValidationActionRedirectPath(credentialToken: string): string {
+		return `saml/${credentialToken}?saml_idp_credentialToken=${credentialToken}`;
```
Comment (pierre-lehnen-rc) [inline apps/meteor/app/meteor-accounts-saml/server/lib/Utils.ts:134]:
```suggestion
		// the saml_idp_credentialToken param is needed by the mobile app
		return `saml/${credentialToken}?saml_idp_credentialToken=${credentialToken}`;
```

**Review by sampaiodiego — DISMISSED** [review]

**Review by casalsgh — DISMISSED** [review]
Approving here based on Diego Sampaio having approved it before and no significant changes happening afterwards (just conflicts and due to me moving PR from develop to 6.3.0 branch).

**Review by sampaiodiego — CHANGES_REQUESTED** [review]
I just realized it is missing a changeset

**Comment by pierre-lehnen-rc** [comment]
> I just realized it is missing a changeset

I've added the changeset now. Also tested with all platforms to ensure the problem is fixed.

**Review by sampaiodiego — APPROVED** [review]

**Pull request merged**